// trpc/lib/src/error.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.TRPCError = exports.TRPCErrorCode = void 0;
///////////////////////
/////  ERROR DEF  /////
///////////////////////
var TRPCErrorCode;
(function (TRPCErrorCode) {
    TRPCErrorCode["InvalidEndpoint"] = "InvalidEndpoint";
    TRPCErrorCode["InvalidMethod"] = "InvalidMethod";
    TRPCErrorCode["InvalidArguments"] = "InvalidArguments";
    TRPCErrorCode["InvalidReturnType"] = "InvalidReturnType";
    TRPCErrorCode["InvalidPath"] = "InvalidPath";
    TRPCErrorCode["EndpointNotFound"] = "EndpointNotFound";
    TRPCErrorCode["NotAuthorized"] = "NotAuthorized";
    TRPCErrorCode["AuthorizationError"] = "AuthorizationError";
    TRPCErrorCode["NameConflict"] = "NameConflict";
    TRPCErrorCode["UnknownError"] = "UnknownError";
})(TRPCErrorCode = exports.TRPCErrorCode || (exports.TRPCErrorCode = {}));
// export type TRPCErrorCode = typeof TRPCErrorCode[keyof typeof TRPCErrorCode];
var TRPCError = /** @class */ (function (_super) {
    __extends(TRPCError, _super);
    // type: TRPCErrorCode;
    function TRPCError(code, 
    // type: TRPCErrorCode,
    data) {
        var _this = _super.call(this, data.message) || this;
        _this.code = code;
        _this.data = data;
        return _this;
        // this.type = type;
    }
    return TRPCError;
}(Error));
exports.TRPCError = TRPCError;
